### PR TITLE
Metadata filtering

### PR DESCRIPTION
**Pull request by mraspaud**
This adds the possibility to use metadata to filter data to load, eg:
`glbl = Scene(service='0° Service', sensor="seviri", reader="hrit_msg", start_time=tslot, end_time=tslot + dt.timedelta(minutes=5), base_dir=thedir)`
which would filter on `service` if it is defined in the filehandler metadata

### CODE REVIEW

File: satpy/readers/yaml_reader.py [hunk 1]
```diff
@@ -543,8 +560,8 @@ def new_filehandlers_for_filetype(self, filetype_info, filenames):
         filehandler_iter = self.new_filehandler_instances(filetype_info,
                                                           filename_iter)
         return [fhd
-                for fhd in self.filter_fh_by_area(self.filter_fh_by_time(
-                    filehandler_iter))]
+                for fhd in self.filter_fh_by_area(self.filter_fh_by_time(self.filter_fh_by_mda(
```
Comment (mraspaud) [inline satpy/readers/yaml_reader.py]: so like
```filtered = self.filter_fh_by_area(self.filter_fh_by_time(filehandler_iter))
return filtered```
?
We could probably return the iterator btw

Comment (mraspaud) [inline satpy/readers/yaml_reader.py]: No we can't

Comment (davidh-ssec) [inline satpy/readers/yaml_reader.py]: I mean:
```
filtered_time = self.filter_fh_by_time(filehandler_iter)
filtered_area = self.filter_fh_by_area(filtered_time)
return list(filtered_area)
```

Or something similar.

File: satpy/readers/yaml_reader.py [hunk 2]
```diff
@@ -381,7 +381,9 @@ def __init__(self,
                  start_time=None,
                  end_time=None,
                  area=None,
-                 filter_filenames=True, **kwargs):
```
Comment (mraspaud) [inline satpy/readers/yaml_reader.py]: ok, putting it back.

**Pull request merged**